examples/lustre/test.bats

```diff
@@ -0,0 +1,76 @@
+load "${CHTEST_DIR}/common.bash"
+
+setup () {
+    scope full
+    prerequisites_ok lustre
+
+    if [[ $SLURM_JOB_ID ]]; then
+        if [[ -z $ch_lustre ]]; then 
+            pedantic_fail 'no lustre to bind mount'
+        fi
+    else
+        if [[ -z $ch_lustre ]]; then
+            skip 'no lustre to bind mount'
+        fi
+    fi
+
+    # Check lustre directory exists
+    if [[ ! -e $ch_lustre ]]; then
+        echo "${ch_lustre} does not exist"
+        exit 1
+    fi
+    # Check lustre directory is a directory
+    if [[ ! -d $ch_lustre ]]; then
+        echo "${ch_lustre} is not a directory"
+        exit 1
+    fi
+
+    binds=${ch_lustre}:/mnt/0
+    work_dir=/mnt/0/charliecloud_testing_directory
+    mkdir -p "${ch_lustre}/charliecloud_testing_directory"
+}
+
+@test "${ch_tag}/create dir" {
+    ch-run -b "$binds" "$ch_img" -- mkdir "${work_dir}/tryme"
+}
+
+@test "${ch_tag}/create file" {
+    ch-run -b "$binds" "$ch_img" -- touch "${work_dir}/test_create.txt"
+}
+
+
+@test "${ch_tag}/delete file" {
+    ch-run -b "$binds" "$ch_img" -- rm -f "${work_dir}/test_create.txt"
+}
+
+@test "${ch_tag}/write" {
+    ch-run -b "$binds" "$ch_img" -- sh -c "sleep 5 && echo hello | dd of=${work_dir}/test_write.txt"
+}
+
+@test "${ch_tag}/read" {
+    out=$(ch-run -b "$binds" "$ch_img" -- dd if="${work_dir}/test_write.txt" iflag=nocache 2> /dev/null)
```

Comment:
Use Bats' `run` command for this. The idiom is to check the return code is what you expect, print the output, and then compare the actual output with expected, including stderr, using `diff -u`, which will fail if there is a difference. There should be examples elsewhere in the code to work from.

Reply:
Turns out that won't work here. `dd` displays the speed at which data was moved which will vary on every run, so there is no way to match the output of stderr here.

Reply:
I can dump the standard error but that defeats the purpose of using the diff idiom here.

Reply:
If there's not too much varying output, you can do something like:
```
$ cat <<'EOF' | diff -u - <(echo "$output" | sed -E s'/(copied),.*$/\1/')
0+1 records in
0+1 records out
234 bytes copied
EOF
--- -	2020-02-26 15:46:21.656131355 -0700
+++ /dev/fd/63	2020-02-26 15:46:21.651184648 -0700
@@ -1,3 +1,3 @@
 0+1 records in
 0+1 records out
-234 bytes copied
+235 bytes copied
```

Reply:
Turns out there is a flag to drop the transfer speed on dd